net/raft/raft_test.go

```diff
@@ -264,6 +269,47 @@ func TestLeaderEviction(t *testing.T) {
 	}
 }
 
+func TestSnapshot(t *testing.T) {
+	ctx := context.Background()
+
+	// Create new test cluster, triggers snapshot
+	node := newSnapshotTestNode(ctx, 2, t)
+	defer node.cleanup()
+
+	// Wait for snapshot to successfully write, times out after 2 seconds
+loop:
+	for {
+		select {
+		case <-time.After(2 * time.Second):
```

Comment:
I think we need to move the `time.After(2 * time.Second)` above the for loop and save it to a local variable. Otherwise each iteration will invoke time.After and receive a new channel, and we'll always hit the default case.